code/bmpman/bmpman.cpp

```diff
@@ -1594,10 +1594,15 @@ int bm_load_animation(const char *real_filename, int *nframes, int *fps, int *ke
 			img_size = the_apng.imgsize();
 		}
 		catch (const apng::ApngException& e) {
-			mprintf(("Failed to load apng: %s\n", e.what() ));
 			if (img_cfp != nullptr)
 				cfclose(img_cfp);
-			return -1;
+			if (rethrow_exceptions) {
+				throw e;
+			}
+			else {
+				mprintf(("Failed to load apng: %s\n", e.what()));
```

Comment:
Add the file name to the log message

Reply:
The file name is already included in the exception message. See `_apng_failed()` in pngutils.cpp.

Reply:
Oh.  Well then, is it also (redundantly) included in the exception messsage on line 4484?

Reply:
In ship.cpp? It is. My reasoning for explicitly including it there was that that call site could, in the future, catch exceptions other than `ApngException`, and those future exceptions might not include the file name in the exception message. Thus I figured it's better to tolerate the redundancy in that case.

Reply:
Gotcha